File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/events/DefaultEventPriority.java
package dev.robocode.tankroyale.botapi.events;

/**
 * Default event priorities values. The higher value, the higher event priority. So the {@link WonRoundEvent} has the
 * highest priority ({@value WON_ROUND}), and {@link DeathEvent} has the lowest priority ({@value DEATH}).
 */
public final class DefaultEventPriority {

    // Hide constructor to prevent instantiation
    private DefaultEventPriority() {
    }

    /**
     * Event priority for the {@link WonRoundEvent}
     */
    public static final int WON_ROUND = 150;
    /**
     * Event priority for the {@link SkippedTurnEvent}
     */
    public static final int SKIPPED_TURN = 140;
    /**
     * Event priority for the {@link TickEvent}
     */
    public static final int TICK = 130;
    /**
     * Event priority for the {@link CustomEvent}
     */
    public static final int CUSTOM = 120;
    /**
     * Event priority for the {@link TeamMessageEvent}.
     */
    public static final int TEAM_MESSAGE = 110;
    /**
     * Event priority for the {@link BotDeathEvent}
     */
    public static final int BOT_DEATH = 100;
    /**
     * Event priority for the {@link BulletHitWallEvent}
     */
    public static final int BULLET_HIT_WALL = 90;
    /**
     * Event priority for the {@link BulletHitBulletEvent}
     */
    public static final int BULLET_HIT_BULLET = 80;
    /**
     * Event priority for the {@link BulletHitBotEvent}
     */
    public static final int BULLET_HIT_BOT = 70;
    /**
     * Event priority for the {@link BulletFiredEvent}
     */
    public static final int BULLET_FIRED = 60;
    /**
     * Event priority for the {@link HitByBulletEvent}
     */
    public static final int HIT_BY_BULLET = 50;
    /**
     * Event priority for the {@link HitWallEvent}
     */
    public static final int HIT_WALL = 40;
    /**
     * Event priority for the {@link HitBotEvent}
     */
    public static final int HIT_BOT = 30;
    /**
     * Event priority for the {@link ScannedBotEvent}
     */
    public static final int SCANNED_BOT = 20;
    /**
     * Event priority for the {@link DeathEvent}
     */
    public static final int DEATH = 10;
}

File: bot-api/dotnet/api/src/internal/DefaultEventPriority.cs
namespace Robocode.TankRoyale.BotApi.Events;

/// <summary>
/// Default event priorities values. The lower value, the higher event priority. So the <see cref="DeathEvent"/>
/// has the highest priority (10), and <see cref="WonRoundEvent"/> has the lowest priority (150).
/// </summary>
public static class DefaultEventPriority
{
    /// <summary>
    /// Event priority for the <see cref="WonRoundEvent"/>
    /// </summary>
    public const int WonRound = 150;

    /// <summary>
    /// Event priority for the <see cref="SkippedTurnEvent"/>
    /// </summary>
    public const int SkippedTurn = 140;

    /// <summary>
    /// Event priority for the <see cref="TickEvent"/>
    /// </summary>
    public const int Tick = 130;

    /// <summary>
    /// Event priority for the <see cref="CustomEvent"/>
    /// </summary>
    public const int Custom = 120;

    /// <summary>
    /// Event priority for the <see cref="TeamMessageEvent"/>
    /// </summary>
    public const int TeamMessage = 110;
    
    /// <summary>
    /// Event priority for the <see cref="BotDeathEvent"/>
    /// </summary>
    public const int BotDeath = 100;

    /// <summary>
    /// Event priority for the <see cref="BulletHitWallEvent"/>
    /// </summary>
    public const int BulletHitWall = 90;

    /// <summary>
    /// Event priority for the <see cref="BulletHitBulletEvent"/>
    /// </summary>
    public const int BulletHitBullet = 80;
    
    /// <summary>
    /// Event priority for the <see cref="BulletHitBotEvent"/>
    /// </summary>
    public const int BulletHitBot = 70;

    /// <summary>
    /// Event priority for the <see cref="BulletFiredEvent"/>
    /// </summary>
    public const int BulletFired = 60;

    /// <summary>
    /// Event priority for the <see cref="HitByBulletEvent"/>
    /// </summary>
    public const int HitByBullet = 50;

    /// <summary>
    /// Event priority for the <see cref="HitWallEvent"/>
    /// </summary>
    public const int HitWall = 40;
    
    /// <summary>
    /// Event priority for the <see cref="HitBotEvent"/>
    /// </summary>
    public const int HitBot = 30;

    /// <summary>
    /// Event priority for the <see cref="ScannedBotEvent"/>
    /// </summary>
    public const int ScannedBot = 20;

    /// <summary>
    /// Event priority for the <see cref="DeathEvent"/>
    /// </summary>
    public const int Death = 10;
}